core/lib/pseudopattern_hunter.js

```diff
@@ -1,81 +1,110 @@
 "use strict";
 
-var pseudopattern_hunter = function () {
-
-  function findpseudopatterns(currentPattern, patternlab) {
-    var glob = require('glob'),
-      fs = require('fs-extra'),
-      pa = require('./pattern_assembler'),
-      lh = require('./lineage_hunter'),
-      Pattern = require('./object_factory').Pattern,
-      plutils = require('./utilities'),
-      path = require('path');
-
-
-    var pattern_assembler = new pa();
-    var lineage_hunter = new lh();
-    var paths = patternlab.config.paths;
-
-    //look for a pseudo pattern by checking if there is a file containing same
-    //name, with ~ in it, ending in .json
-    var needle = currentPattern.subdir + '/' + currentPattern.fileName + '~*.json';
-    var pseudoPatterns = glob.sync(needle, {
-      cwd: paths.source.patterns,
-      debug: false,
-      nodir: true
-    });
-
-    if (pseudoPatterns.length > 0) {
-      for (var i = 0; i < pseudoPatterns.length; i++) {
+const glob = require('glob'),
+  fs = require('fs-promise'),
+  lh = require('./lineage_hunter'),
+  Pattern = require('./object_factory').Pattern,
+  plutils = require('./utilities'),
+  path = require('path');
+
+
+function promiseGlobMatches(currentPattern, paths) {
+  //look for a pseudo pattern by checking if there is a file containing same
+  //name, with ~ in it, ending in .json
+  const needle = currentPattern.subdir + '/' + currentPattern.fileName + '~*.json';
+
+  return new Promise((resolve, reject) => {
+    glob(
+      needle,
+      { cwd: paths.source.patterns, debug: false, nodir: true },
+      function (err, matches) {
+        if (err) { reject(err); }
+        resolve(matches);
+      }
+    );
+  });
+}
+
+function createPseudoPatternObject(variantFileData, pattern, pseudoPattern, patternlab) {
```

Comment:
I really like what you did with this file to make each function have a clearer separation of concerns, but not to a zealous extent